extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MSQTaskQueryMakerUtils.java

```diff
@@ -100,7 +100,7 @@ public static void validateContextSortOrderColumnsExist(
    */
   public static void validateRealtimeReindex(QueryContext context, MSQDestination destination, Query<?> query)
   {
-    final SegmentSource segmentSources = MultiStageQueryContext.getSegmentSources(context);
+    final SegmentSource segmentSources = MultiStageQueryContext.getSegmentSources(context, SegmentSource.NONE);
```

Comment:
`SegmentSource.NONE` is now hard-coded in two places, let's make it just one. You could do a `public static final SegmentSource DEFAULT_SEGMENT_SOURCE` in `IndexerControllerContext`, and reference that here.